doc/md/versioned/troubleshoot.mdx

```diff
@@ -0,0 +1,289 @@
+migration directory, this case happens due to a misconfiguration, e.g. not executing the linter before applying the
+migration or applying against the wrong dialect (like SQLite files against MySQL).
+- **Schema Dependant Change**: One SQL statement was incorrect in schema context, e.g. adding a table that does already
+exist. This case is usually caused by a schema drift, e.g. because someone manually manipulated the database schema.
+- **Data Dependant Change**: One SQL statement contained a data dependant change. If your migration adds a constraint
+like `NOT NULL` and the column does contain `NULL` values, the migration will throw an error. Though Atlas' linting
+will warn if such a change is detected, this is still a possible scenario that could break migration execution.
+- **Connection Loss**: For a number of reasons the connection to the database could be interrupted, e.g. due to a
+network failure or database crash.
+
+### Connection Loss Failures
+
+Just re-try migration execution. Atlas is smart enough to figure out where to proceed execution. For very rare cases,
+where when working without a transaction (or after an implicit commit) a connection loss occurs between an SQL statement
+and the entry in Atlas schema history table, you might need to manually revert the last applied statement because in
+those cases Atlas will continue applying one statement early.
+
+### Syntax Errors
+
+Fix the migration file and re-try migration execution. Atlas is smart enough to continue the execution starting with the
+statement following the last successfully applied one.
+
+### DDL Semantics
+
+Creating or editing schema resources can result in migration failure, e.g. if you are trying to create a table that does
+already exist or to modify a table that does not exist. The cause for this is either an incorrect migration file or a
+schema drift, e.g. if someone dropped, created or modified a resource.
+
+#### Option 1: Incorrect Migration File
+
+Solution can be simple: fix the migration file and re-apply the migration. For cases with multiple deployments or no
+(fast) editorial access to the migration file you have to fix the database state to match whatever the migration file is
+expecting. You can either do this by hand or use one of the `atlas schema` commands. See
+[Option 2](#option-2-schema-drift) for more information.
+
+Once the migration / database is fixed, you can retry migration execution.
+
+#### Option 2: Schema Drift
+
+In case of a schema drift, we still consider the migration files the source of truth. Editing its contents to match the
+managed database will most likely break other deployments or the local dev. The solution is to fix the database schema.
+This can be done manually or with the help of `atlas schema apply` / `atlas schema diff` commands. Both commands know
+how to read a database state from a migration directory and compare it with a state read from a database. The only
```

Comment:
```suggestion
This can be done manually or with the help of `atlas schema apply` / `atlas schema diff` commands. Both commands can
read a database state from a migration directory and compare it with the current state of the target database. The only
```